crates/toml_writer/src/integer.rs

```diff
@@ -0,0 +1,194 @@
+use core::fmt::{self, Display};
+
+/// N must be an integer type.
+#[derive(Copy, Clone, Debug)]
+pub struct TomlInteger<N> {
+    value: N,
+    radix: Radix,
+}
+
+impl crate::WriteTomlValue for TomlInteger<u8> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<i8> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<u16> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<i16> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<u32> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<i32> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<u64> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<i64> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<u128> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<i128> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<usize> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+impl crate::WriteTomlValue for TomlInteger<isize> {
+    fn write_toml_value<W: crate::TomlWrite + ?Sized>(&self, writer: &mut W) -> fmt::Result {
+        write_toml_value(self.value, &self.radix, writer)
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+pub struct TomlIntegerFormat {
+    radix: Radix,
+    _separators: (), // Placeholder for future use
+}
```

Comment:
As this is the primary type people will interact with, please put this at the top of the file